src/software/geom/algorithms/step_along_perimeter.cpp

```diff
@@ -0,0 +1,72 @@
+#include "software/geom/algorithms/step_along_perimeter.h"
+
+#include <algorithm>
+#include <vector>
+
+#include "software/geom/algorithms/collinear.h"
+#include "software/geom/algorithms/distance.h"
+#include "software/geom/segment.h"
+
+
+Point stepAlongPerimeter(const Polygon& polygon, const Point& start,
+                         double travelDistance)
```

Comment:
refer to our codestyle guide: https://github.com/UBC-Thunderbots/Software/blob/master/docs/code-style-guide.md

variables are kept lower_case